### PR TITLE
feat(clustering) register wrpc service from PDK

### PR DESCRIPTION
**Pull request by Suika-Kong**
Add `kong.cluster.service.register` and `kong.cluster.service.get_services`, so we can add wRPC service in plugins via PDK APIs.

fix FT-3048

### CODE REVIEW

File: kong/clustering/init.lua [hunk 1]
```diff
@@ -6,6 +6,9 @@ local pl_file = require("pl.file")
 local pl_tablex = require("pl.tablex")
 local ssl = require("ngx.ssl")
 local openssl_x509 = require("resty.openssl.x509")
+
+local services_init = require "kong.clustering.services".init
```
Comment (chronolaw) [inline kong/clustering/init.lua:10]: should use `require(...).init` to keep style clean.

File: kong/clustering/services/config.lua [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  end)
+
+  wrpc_service:set_handler("ConfigService.ReportMetadata", function(peer, data)
+    local client = peer.client
+    if client then
+      ngx_log(INFO, _log_prefix, "received initial metadata package from client: ", client.dp_id)
+      client.basic_info = data
+      client.basic_info_semaphore:post()
+    end
+    return {
```
Comment (chronolaw) [inline kong/clustering/services/config.lua:41]: `{ok = 'done'}` should be a constant table

File: kong/clustering/services/config.lua [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      peer.config_obj.next_hashes = data.hashes
+      peer.config_obj.next_config_version = tonumber(data.version)
+      if peer.config_semaphore:count() <= 0 then
+        -- the following line always executes immediately after the `if` check
+        -- because `:count` will never yield, end result is that the semaphore
+        -- count is guaranteed to not exceed 1
+        peer.config_semaphore:post()
+      end
+    end
+    return { accepted = true }
```
Comment (chronolaw) [inline kong/clustering/services/config.lua:70]: `{ accepted = true }` should be a constant table.

File: kong/clustering/services/init.lua [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+local function register(name, versions, service_init_dp, service_init_cp)
+  if supported[name] then
+    error("service name conflict or registered multiple times: " .. name)
+  end
+
+  if not versions[1] then
+    error("at least one version should be provided for service " .. name)
+  end
+
+  for i, version in ipairs(versions) do
```
Comment (chronolaw) [inline kong/clustering/services/init.lua:49]: localize `ipairs`

File: kong/clustering/services/init.lua [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+-- get all cluster services
+---@return table services
+function _M.get_services()
+  return supported
+end
+
+function _M.init(conf)
+  -- otherwise it would be recusive reference
+  require "kong.clustering.services.negotiation".init(conf, dp_service, cp_service)
```
Comment (chronolaw) [inline kong/clustering/services/init.lua:77]: use style `require(...)`

File: kong/pdk/cluster.lua [hunk 6]
```diff
@@ -5,6 +5,9 @@
 
 local kong = kong
 local CLUSTER_ID_PARAM_KEY = require("kong.constants").CLUSTER_ID_PARAM_KEY
+local clustering_services = require "kong.clustering.services"
```
Comment (chronolaw) [inline kong/pdk/cluster.lua:8]: require style

File: kong/clustering/services/config.lua (hunk 2 above)
Comment (Suika-Kong) [reply kong/clustering/services/config.lua:41]: Yes. That's a remaining from the last PR.

**Comment by Suika-Kong** [comment]
This PR is outdated due to updating on wrpc.

**Pull request closed without merging**